packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -71,26 +73,39 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
     SignInOption signInOption = SignInOption.standard,
     String? hostedDomain,
     String? clientId,
-  }) async {
-    final String? appClientId = clientId ?? _autoDetectedClientId;
+  }) {
+    return initWithParams(SignInInitParameters(
+      scopes: scopes,
+      signInOption: signInOption,
+      hostedDomain: hostedDomain,
+      clientId: clientId,
+    ));
+  }
+
+  @override
+  Future<void> initWithParams(SignInInitParameters params) async {
+    final String? appClientId = params.clientId ?? _autoDetectedClientId;
     assert(
         appClientId != null,
         'ClientID not set. Either set it on a '
         '<meta name="google-signin-client_id" content="CLIENT_ID" /> tag,'
-        ' or pass clientId when calling init()');
+        ' or pass clientId when calling initialize()');
+
+    assert(params.serverClientId == null,
+        'serverClientId is not supported on Web.');
```

Comment:
I guess that if we're in this case, the authentication on the web would probably not work as expected at all? So an assert is probably OK. Can we add some documentation to the README describing this feature disparity?

Reply:
Done